test/topology_custom/test_multidc.py

```diff
@@ -0,0 +1,209 @@
+    conn.execute(f"USE {ks}")
+    conn.execute(
+        f"CREATE COLUMNFAMILY {cf} ( key varchar, c varchar, v varchar, PRIMARY KEY(key, c)) WITH comment='test cf'"
+    )
+
+    logger.info("Perform insert/overwrite")
+    putget(conn, cf, ks)
+
+
+@pytest.mark.asyncio
+async def test_putget_2dc_rf2(
+        request: pytest.FixtureRequest, manager: ManagerClient
+) -> None:
+    """Simple put-get test for 2 DC with 2 node each (RF=2) -- tests cross-DC efficient writes"""
+    ks = "test_ks"
+    cf = "test_cf"
+    table_name = "test_table_name"
+    columns = [Column("name", TextType), Column("value", TextType)]
+    logger.info("Create two servers in different DC's")
+    for i in [1, 1, 2, 2]:
+        s_info = await manager.server_add(
+            config=CONFIG,
+            property_file={"dc": f"dc{i}", "rack": "myrack"},
+        )
+        logger.info(s_info)
+    conn = manager.get_cql()
+    logger.info(conn.execute('desc cluster').one())
+    random_tables = RandomTables(request.node.name, manager, ks, 2)
+    await random_tables.add_table(ncolumns=2, columns=columns, pks=1, name=table_name)
+    conn.execute(f"USE {ks}")
+    conn.execute(
+        f"CREATE COLUMNFAMILY {cf} ( key varchar, c varchar, v varchar, PRIMARY KEY(key, c)) WITH comment='test cf'"
+    )
+
+    logger.info("Perform insert/overwrite")
+    putget(conn, cf, ks)
+
+
+@pytest.mark.asyncio
+async def test_query_dc_with_rf_0_does_not_crash_db(request: pytest.FixtureRequest, manager: ManagerClient):
+    """Test querying dc with CL=LOCAL_QUORUM when RF=0 for this dc, does not crash the node and returns None
+    Covers https://github.com/scylladb/scylla/issues/8354"""
+    servers = []
+    ks = "test_ks"
+    table_name = "test_table_name"
+    expected = ["k1", "value1"]
+    dc_replication = {'dc2': 0}
+    columns = [Column("name", TextType), Column("value", TextType)]
+
+    for i in [1, 2]:
+        servers.append(await manager.server_add(
+            config=CONFIG,
+            property_file={"dc": f"dc{i}", "rack": "myrack"},
+        ))
+
+    dc1_connection = manager.connect_to_node(servers[0])
+    dc2_connection = manager.connect_to_node(servers[1])
+    random_tables = RandomTables(request.node.name, manager, ks, 1, dc_replication)
+    await random_tables.add_table(ncolumns=2, columns=columns, pks=1, name=table_name)
+    dc1_connection.execute(
+        f"INSERT INTO  {ks}.{table_name} ({columns[0].name}, {columns[1].name}) VALUES ('{expected[0]}', '{expected[1]}');")
+    query_cl_one = SimpleStatement(f"SELECT * from {ks}.{table_name};", consistency_level=ConsistencyLevel.ONE)
+    query_cl_local_quorum = SimpleStatement(f"SELECT * from {ks}.{table_name};",
+                                            consistency_level=ConsistencyLevel.LOCAL_QUORUM)
+    nodetool.flush(dc1_connection, "{ks}.{table_name}")
+    first_node_results = list(dc1_connection.execute(query_cl_one).one())
```

Comment:
I know this test is copied from the existin dtest, so I can't complain, but I'm still curious why it was important to query DC1 with CL=ONE... If the bug we were hunting was about LOCAL_QUORUM on DC2, doesn't it make sense to also query DC1 with LOCAL_QUORUM?